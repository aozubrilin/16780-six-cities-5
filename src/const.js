export const STAR_TO_PERSENT = 20;

export const OfferCardClass = {
  MAIN: `cities`,
  NEAR: `near-places`,
  FAVORITE: `favorites`,
};

export const CITIES = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export const SortType = {
  POPULAR: `Popular`,
  TO_HIGHT: `Price: low to high`,
  TO_LOW: `Price: high to low`,
  TOP_RATE: `Top rated first`,
};
