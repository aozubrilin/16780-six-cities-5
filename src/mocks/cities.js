export const CityCoordinate = {
  BRUSSELS: {coordinates: [50.85, 4.3]},
  PARIS: {coordinates: [48.86, 2.34]},
  AMSTERDAM: {coordinates: [52.38333, 4.9]},
  COLOGNE: {coordinates: [50.933, 6.95]},
  HAMBURG: {coordinates: [53.56, 10.015]},
  DUSSELDORF: {coordinates: [51.22, 6.78]},
};
