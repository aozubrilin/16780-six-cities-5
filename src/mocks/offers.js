export default [
  {
    id: 0,
    city: `Amsterdam`,
    title: `Beautiful & luxurious apartment at great location`,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    type: `Apartment`,
    images: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/room.jpg`
    ],
    rating: 1,
    bedroomsCount: 1,
    adultsCount: 1,
    price: 100,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`
    ],
    host: {
      name: `Max`,
      avatar: `img/avatar-max.jpg`,
      isProUser: true,
    },
    coordinates: [52.3909553943508, 4.85309666406198],
    isPremium: true,
    isFavorite: true,
  },
  {
    id: 1,
    city: `Cologne`,
    title: `Wood and stone place`,
    description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    type: `Room`,
    images: [
      `img/apartment-02.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-02.jpg`,
    ],
    rating: 2,
    bedroomsCount: 2,
    adultsCount: 2,
    price: 200,
    features: [
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    host: {
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`,
      isProUser: false,
    },
    coordinates: [50.9381, 6.95694],
    isPremium: false,
    isFavorite: false,
  },
  {
    id: 2,
    city: `Brussels`,
    title: `Canal View Prinsengracht`,
    description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    type: `House`,
    images: [
      `img/apartment-03.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`
    ],
    rating: 3,
    bedroomsCount: 3,
    adultsCount: 3,
    price: 300,
    features: [
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`
    ],
    host: {
      name: `Ann`,
      avatar: `img/avatar.svg`,
      isProUser: false,
    },
    coordinates: [50.80, 4.347],
    isPremium: true,
    isFavorite: false,
  },
  {
    id: 3,
    city: `Dusseldorf`,
    title: `White castle`,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    type: `Hotel`,
    images: [
      `img/room.jpg`,
      `img/apartment-03.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/apartment-01.jpg`
    ],
    rating: 4,
    bedroomsCount: 4,
    adultsCount: 4,
    price: 400,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`
    ],
    host: {
      name: `Bob`,
      avatar: `img/avatar.svg`,
      isProUser: true,
    },
    coordinates: [51.22, 6.78],
    isPremium: false,
    isFavorite: true,
  },
  {
    id: 5,
    city: `Amsterdam`,
    title: `Canal View Prinsengracht`,
    description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    type: `House`,
    images: [
      `img/apartment-03.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`
    ],
    rating: 3,
    bedroomsCount: 3,
    adultsCount: 3,
    price: 500,
    features: [
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`
    ],
    host: {
      name: `Ann`,
      avatar: `img/avatar.svg`,
      isProUser: false,
    },
    coordinates: [52.369553943508, 4.85309666406198],
    isPremium: true,
    isFavorite: false,
  },
  {
    id: 6,
    city: `Amsterdam`,
    title: `White castle`,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    type: `Hotel`,
    images: [
      `img/room.jpg`,
      `img/apartment-03.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/apartment-01.jpg`
    ],
    rating: 4,
    bedroomsCount: 4,
    adultsCount: 4,
    price: 600,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`
    ],
    host: {
      name: `Bob`,
      avatar: `img/avatar.svg`,
      isProUser: true,
    },
    coordinates: [52.3909553943508, 4.929309666406198],
    isPremium: false,
    isFavorite: true,
  },
  {
    id: 7,
    city: `Amsterdam`,
    title: `Canal View Prinsengracht`,
    description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    type: `House`,
    images: [
      `img/apartment-03.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`
    ],
    rating: 3,
    bedroomsCount: 3,
    adultsCount: 3,
    price: 300,
    features: [
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`
    ],
    host: {
      name: `Ann`,
      avatar: `img/avatar.svg`,
      isProUser: false,
    },
    coordinates: [52.3809553943508, 4.939309666406198],
    isPremium: true,
    isFavorite: false,
  },
];
