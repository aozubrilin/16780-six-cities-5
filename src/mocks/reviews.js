export default [
  {
    id: 0,
    name: `Max`,
    avatar: `img/avatar-max.jpg`,
    reiting: 1,
    date: new Date(`2019-04-18`),
    text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Phasellus egestas tellus rutrum tellus.`,
  },
  {
    id: 1,
    name: `Angelina`,
    avatar: `img/avatar-angelina.jpg`,
    reiting: 2,
    date: new Date(`2020-06-18`),
    text: `Volutpat lacus laoreet non curabitur gravida arcu. Morbi blandit cursus risus at ultrices mi tempus. Consectetur adipiscing elit pellentesque habitant morbi. Fermentum leo vel orci porta.`,
  },
  {
    id: 2,
    name: `Ann`,
    avatar: `img/avatar.svg`,
    reiting: 3,
    date: new Date(`2020-06-17`),
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
  },
  {
    id: 3,
    name: `Bob`,
    avatar: `img/avatar.svg`,
    reiting: 4,
    date: new Date(`2020-09-17`),
    text: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
  }
];
