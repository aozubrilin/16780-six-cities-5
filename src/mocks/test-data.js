export const offers = [
  {
    bedroomsCount: 3,
    city: {
      name: `Brussels`,
      coordinates: [50.846557, 4.351697],
      zoom: 13
    },
    description: `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
    features: [`Towels`, `Baby seat`, `Laptop friendly workspace`, `Dishwasher`, `Washing machine`, `Air conditioning`, `Washer`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isProUser: true,
      name: `Angelina`
    },
    id: 2,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    ],
    isFavorite: false,
    isPremium: false,
    coordinates: [50.839557, 4.346697],
    adultsCount: 8,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
    price: 323,
    rating: 4.3,
    title: `Amazing and Extremely Central Flat`,
    type: `Apartment`
  },
  {
    bedroomsCount: 4,
    city: {
      name: `Brussels`,
      coordinates: [50.846557, 4.351697],
      zoom: 13
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    features: [`Washer`, `Laptop friendly workspace`, `Breakfast`, `Air conditioning`, `Baby seat`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isProUser: true,
      name: `Angelina`
    },
    id: 3,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    ],
    isFavorite: false,
    isPremium: false,
    coordinates: [50.865556999999995, 4.371696999999999],
    adultsCount: 7,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    price: 218,
    rating: 4.6,
    title: `The house among olive `,
    type: `Apartment`
  },
  {
    bedroomsCount: 1,
    city: {
      name: `Amsterdam`,
      coordinates: [52.37454, 4.897976],
      zoom: 13
    },
    description: `Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.`,
    features: [`Breakfast`, `Air conditioning`, `Laptop friendly workspace`, `Baby seat`, `Washer`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isProUser: true,
      name: `Angelina`
    },
    id: 22,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`
    ],
    isFavorite: false,
    isPremium: true,
    coordinates: [52.364540000000005, 4.9019759999999994],
    adultsCount: 3,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
    price: 175,
    rating: 3.9,
    title: `Beautiful & luxurious apartment at great location`,
    type: `Room`,
  }
];

export const favoriteOffers = [
  {
    bedroomsCount: 3,
    city: {
      name: `Brussels`,
      coordinates: [50.846557, 4.351697],
      zoom: 13
    },
    description: `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
    features: [`Towels`, `Baby seat`, `Laptop friendly workspace`, `Dishwasher`, `Washing machine`, `Air conditioning`, `Washer`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isProUser: true,
      name: `Angelina`
    },
    id: 2,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    ],
    isFavorite: true,
    isPremium: false,
    coordinates: [50.839557, 4.346697],
    adultsCount: 8,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
    price: 323,
    rating: 4.3,
    title: `Amazing and Extremely Central Flat`,
    type: `Apartment`
  },
  {
    bedroomsCount: 4,
    city: {
      name: `Brussels`,
      coordinates: [50.846557, 4.351697],
      zoom: 13
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    features: [`Washer`, `Laptop friendly workspace`, `Breakfast`, `Air conditioning`, `Baby seat`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isProUser: true,
      name: `Angelina`
    },
    id: 3,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    ],
    isFavorite: true,
    isPremium: false,
    coordinates: [50.865556999999995, 4.371696999999999],
    adultsCount: 7,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    price: 218,
    rating: 4.6,
    title: `The house among olive `,
    type: `Apartment`
  },
  {
    bedroomsCount: 1,
    city: {
      name: `Amsterdam`,
      coordinates: [52.37454, 4.897976],
      zoom: 13
    },
    description: `Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.`,
    features: [`Breakfast`, `Air conditioning`, `Laptop friendly workspace`, `Baby seat`, `Washer`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isProUser: true,
      name: `Angelina`
    },
    id: 22,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`
    ],
    isFavorite: true,
    isPremium: true,
    coordinates: [52.364540000000005, 4.9019759999999994],
    adultsCount: 3,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
    price: 175,
    rating: 3.9,
    title: `Beautiful & luxurious apartment at great location`,
    type: `Room`,
  }
];

export const reviews = [
  {
    id: 1,
    date: `2020-11-05T13:38:44.753Z`,
    text: `The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.`,
    rating: 4,
    user: {
      id: 16,
      avatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/7.jpg`,
      isProUser: true,
      name: `Mollie`
    }
  },
  {
    id: 2,
    date: `2020-11-05T13:38:44.753Z`,
    text: `I stayed here for one night and it was an unpleasant experience.`,
    rating: 4,
    user: {
      id: 19,
      avatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg`,
      isProUser: false,
      name: `Christina`
    }
  },
  {
    id: 3,
    date: `2020-11-05T13:38:44.753Z`,
    text: `The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.`,
    rating: 5,
    user: {
      id: 12,
      avatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg`,
      isProUser: true,
      name: `Isaac`
    }
  }
];

export const mockStore = {
  app: {
    city: `Amsterdam`,
    cities: [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`],
    currentSortType: `Popular`,
    activeOfferId: -1,
  },
  data: {
    offers: [],
    isLoadedOffers: false,
    nearOffers: [],
    isLoadedNearOffers: false,
    favoriteOffers: [],
    isLoadedfavoriteOffers: false,
    currentOffer: {},
    isLoadedCurrentOffer: false,
    errorFetchMessadge: ``,
    reviews: [],
    isLoadedRviews: false,
  },
  user: {
    authorizationStatus: `NO_AUTH`,
    userInfo: {},
  }
};
