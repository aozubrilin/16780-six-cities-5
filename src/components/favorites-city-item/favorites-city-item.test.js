import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {Router} from "react-router-dom";
import history from "../../browser-history";
import FavoritesCityItem from "./favorites-city-item";
import {mockStore, favoriteOffers} from "../../mocks/test-data";
import {CITIES} from "../../const";


describe(`FavoritesCityItem snapshot`, () => {
  const testStore = configureStore();
  const store = testStore(mockStore);

  it(`Should FavoritesCityItem render correctly`, () => {
    const tree = renderer.create(
        <Provider store={store}>
          <Router history={history}>
            <FavoritesCityItem
              city={CITIES[2]}
              offers={favoriteOffers}
            />
          </Router>
        </Provider>
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
