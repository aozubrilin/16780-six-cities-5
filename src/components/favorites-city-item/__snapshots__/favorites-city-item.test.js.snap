// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesCityItem snapshot Should FavoritesCityItem render correctly 1`] = `
<li
  className="favorites__locations-items"
>
  <div
    className="favorites__locations locations locations--current"
  >
    <div
      className="locations__item"
    >
      <a
        className="locations__item-link"
        href="#"
      >
        <span>
          Brussels
        </span>
      </a>
    </div>
  </div>
  <div
    className="favorites__places"
  >
    <article
      className="favorites__card place-card "
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      
      <div
        className="favorites__image-wrapper place-card__image-wrapper"
      >
        <a
          href="/offer/2"
          onClick={[Function]}
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="110"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg"
            width="150"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              323
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <a
            className="place-card__bookmark-button place-card__bookmark-button--active button"
            href="/login"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              In
               bookmarks
            </span>
          </a>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/2"
            onClick={[Function]}
          >
            Amazing and Extremely Central Flat
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Apartment
        </p>
      </div>
    </article>
    <article
      className="favorites__card place-card "
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      
      <div
        className="favorites__image-wrapper place-card__image-wrapper"
      >
        <a
          href="/offer/3"
          onClick={[Function]}
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="110"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
            width="150"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              218
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <a
            className="place-card__bookmark-button place-card__bookmark-button--active button"
            href="/login"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              In
               bookmarks
            </span>
          </a>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/3"
            onClick={[Function]}
          >
            The house among olive 
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Apartment
        </p>
      </div>
    </article>
    <article
      className="favorites__card place-card "
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="favorites__image-wrapper place-card__image-wrapper"
      >
        <a
          href="/offer/22"
          onClick={[Function]}
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="110"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg"
            width="150"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              175
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <a
            className="place-card__bookmark-button place-card__bookmark-button--active button"
            href="/login"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              In
               bookmarks
            </span>
          </a>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/22"
            onClick={[Function]}
          >
            Beautiful & luxurious apartment at great location
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Room
        </p>
      </div>
    </article>
  </div>
</li>
`;
